apps/prairielearn/src/ee/pages/instructorAiGenerateDraftEditor/instructorAiGenerateDraftEditor.ts

```diff
@@ -299,7 +299,24 @@ router.post(
         req.body.qid,
       );
 
-      res.redirect(res.locals.urlPrefix + '/question/' + qid + '/settings');
+      const client = getCourseFilesClient();
+
+      const result = await client.batchDeleteQuestions.mutate({
+        course_id: res.locals.course.id,
+        user_id: res.locals.user.user_id,
+        authn_user_id: res.locals.authn_user.user_id,
+        has_course_permission_edit: res.locals.authz_data.has_course_permission_edit,
+        question_ids: [question.id],
+      });
+
+      if (result.status === 'error') {
+        throw new error.HttpStatusError(
+          500,
+          'Draft deletion failed, but question creation succeeded.',
+        );
+      }
+
+      res.redirect(res.locals.urlPrefix + '/question/' + qid + '/preview');
```

Comment:
Can we also get a flash message saying the new question is now available?

Reply:
Added a flash message in 9741c215b9ab21d261c441ee5082255c757475b9.